pkg/ostree/ostree.go

```diff
@@ -152,7 +152,10 @@ func ResolveRef(location, ref string, consumerCerts bool, subs *rhsm.Subscriptio
 	if consumerCerts {
 		if subs == nil {
 			subs, err = rhsm.LoadSystemSubscriptions()
-			if subs.Consumer == nil || err != nil {
+			if err != nil {
+				return "", NewResolveRefError("error adding rhsm certificates when resolving ref: %s", err)
+			}
+			if subs.Consumer == nil {
 				return "", NewResolveRefError("error adding rhsm certificates when resolving ref")
```

Comment:
> we expect certs but got none and no error

That's not entirely accurate and is where the issue comes from.  The `LoadSystemSubscriptions()` function is "allowed" to return empty consumer certs without an error for systems that have RHSM installed but aren't subscribed.  The depsolver calls this function to get subscription certs to include in depsolve requests.

In the case of ostree, the request for consumer certs is explicit, so we need to check that both (1) the function didn't fail to find RHSM system certs and (2) the consumer certs were also loaded.

It's confusing and strange and I'd like to address it in a separate PR.  We should either make the function fail if it doesn't find consumer certs, because I think there's never a case where we care about RHSM certs but don't care about consumer certs, or, if there really are cases where we care about one and not the other, add an argument or variant of the function (`LoadSystemSubscriptionWithConsumerCerts()`) or `func LoadSystemSubscription(consumerRequired bool)` (ignoring the known issues of bools in fn args).